### PR TITLE
Added `catchError` operator

### PR DESCRIPTION
**Pull request by davidwdan**
Only implemented the selector variant, since that covers all of the use cases

### CODE REVIEW

File: demo/catch/catchError.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+require_once __DIR__ . '/../bootstrap.php';
+
+
```
Comment (asm89) [inline demo/catch/catchError.php]: nit: whitespace

File: test/Rx/Functional/Operator/CatchErrorTest.php [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+                onNext(220, 3),
+                onError(230, $error)
+            ]
+        );
+
+
+        $results = $this->scheduler->startWithCreate(function () use ($o1, $error2) {
+            return $o1->catchError(function () use ($error2) {
+                throw $error2;
+
```
Comment (asm89) [inline test/Rx/Functional/Operator/CatchErrorTest.php:349]: nit: whitespace

File: test/Rx/Functional/Operator/CatchErrorTest.php [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+        $o1 = $this->createHotObservable(
+            [
+
+                onNext(150, 1),
+                onNext(210, 2),
+                onError(215, $error)
+            ]
+        );
+
+
```
Comment (asm89) [inline test/Rx/Functional/Operator/CatchErrorTest.php]: nit: all this whitespace :P

**Comment by asm89** [comment]
Apart from some nits this all looks great to me. Had to bend my mind a few times over the tests, but they seem to make sense.

Can you fix some of the whitespace things + squash?

**Comment by davidwdan** [comment]
@asm89 fixed whitespace and squashed

**Pull request merged**